server/src/main/java/org/elasticsearch/index/mapper/IpFieldMapper.java

```diff
@@ -497,18 +499,14 @@ private static InetAddress value(XContentParser parser, InetAddress nullValue) t
     }
 
     private void indexValue(DocumentParserContext context, InetAddress address) {
+        if (dimension) {
+            // Extract the tsid part of the dimension field
+            BytesReference bytes = TimeSeriesIdFieldMapper.extractTsidValue(NetworkAddress.format(address));
+            context.doc().addDimensionBytes(fieldType().name(), bytes);
```

Comment:
Now that I think about.... Can't you declare something a dimension without enabling the tsid field? I wonder if we should avoid encoding it if the field isn't present.

Reply:
For `ip` and numeric field types, we can skip the `encodeTsidValue` part if `_tsid` field is not going to be generated. 

However, for `keyword` fields we still must do the encoding, because must validate the string length.